### PR TITLE
platform.interfaces: add version 0.3.41, support conan v2

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **platform.interfaces/***

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by danimtb — APPROVED** [review]

File: recipes/platform.interfaces/all/conandata.yml [hunk 1]
```diff
@@ -1,4 +1,7 @@
 sources:
+  "0.3.41":
+    url: "https://github.com/linksplatform/Interfaces/releases/download/cpp_0.3.41/platform.interfaces_0.3.41.zip"
+    sha256: "26bfb96f8918db86390cf326eba5c76cfc29f06a785d30c789445fe59371bab4"
```
Comment (uilianries) [inline recipes/platform.interfaces/all/conandata.yml:4]:
```suggestion
    source:
      url: "https://github.com/linksplatform/Interfaces/releases/download/cpp_0.3.41/platform.interfaces_0.3.41.zip"
      sha256: "26bfb96f8918db86390cf326eba5c76cfc29f06a785d30c789445fe59371bab4"
    license:
      url: "https://raw.githubusercontent.com/linksplatform/Interfaces/cpp_0.3.41/LICENSE"
      sha256: "79d0fc44716007dddc375601bca8879ad45bc1165bf9342b7a16572b4f41abe8"
```

File: recipes/platform.interfaces/all/conandata.yml [hunk 2]
```diff
@@ -1,4 +1,7 @@
...
   "0.1.2":
-    url: https://github.com/linksplatform/Interfaces/archive/refs/tags/0.4.0_0.1.2.zip
-    sha256: 317c29e7ac6122af14e8b1cc8e674a71ba0969c0391ee54d9e045a88d25b5739
+    url: "https://github.com/linksplatform/Interfaces/archive/refs/tags/0.4.0_0.1.2.tar.gz"
+    sha256: "61283bf6488b9051e9f9e80e89ab0d1f85b76a1dbee3fd27f193e678f23903e2"
```
Comment (uilianries) [inline recipes/platform.interfaces/all/conandata.yml:7]:
```suggestion
    source:
      url: "https://github.com/linksplatform/Interfaces/archive/refs/tags/0.4.0_0.1.2.tar.gz"
      sha256: "61283bf6488b9051e9f9e80e89ab0d1f85b76a1dbee3fd27f193e678f23903e2"
```

File: recipes/platform.interfaces/all/conanfile.py [hunk 3]
```diff
@@ -1,60 +1,98 @@
...
+
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(f"{self.ref} requires C++{self._min_cppstd}, "
+                                            "which is not supported "
+                                            "by {self.settings.compiler} {self.settings.compiler.version}.")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
```
Comment (uilianries) [inline recipes/platform.interfaces/all/conanfile.py:62]:
```suggestion
        get(self, **self.conan_data["sources"][self.version]["source"], strip_root=True)
```

File: recipes/platform.interfaces/all/conanfile.py [hunk 4]
```diff
@@ -1,60 +1,98 @@
...
+        if Version(self.version) >= "0.3.41":
+            download(
+                    self,
+                    url="https://raw.githubusercontent.com/linksplatform/Interfaces/cpp_0.3.41/LICENSE",
+                    filename=os.path.join(self.source_folder, "LICENSE"),
+                    sha256="79d0fc44716007dddc375601bca8879ad45bc1165bf9342b7a16572b4f41abe8",
+                    )
```
Comment (uilianries) [inline recipes/platform.interfaces/all/conanfile.py:69]:
```suggestion
            get(self, **self.conan_data["sources"][self.version]["license"], strip_root=True)
```

**Review by uilianries — COMMENTED** [review]
Let's list the license from conandata.yml

**Comment by toge** [comment]
@uilianries 

There is an error:
```
ERROR: platform.interfaces/0.3.41: Error in source() method, line 64
	get(self, **self.conan_data["sources"][self.version]["license"], strip_root=True)
	BadZipFile: File is not a zip file
```

I think we should use `download` instead of `get` for getting the license file.

**Review by uilianries — APPROVED** [review]
LGTM

**Review by prince-chrismc — APPROVED** [review]

**Comment by FreePhoenix888** [comment]
@toge , @uilianries , @prince-chrismc ,Thank you for improvements in conan reciper. I am very grateful for your efforts.

Why is apple-clang support dropped?

**Comment by toge** [comment]
@FreePhoenix888 
Thank you for your comment.
I dropped apple-clang support because the build of apple-clang failed in the conan recipe test code.

However, this was not an appropriate response as you asked.
I will try to modify the test code to scuccess building in apple-clang.

**Comment by toge** [comment]
@FreePhoenix888 
I failed to compile sample program of platform.interfaces.
apple-clang 14 complained following:
```
/Users/toge/.conan/data/platform.interfaces/0.3.41/toge/stable/package/5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9/include/CEnumerable.h:8:40: error: no member named 'range' in namespace 'std::ranges'
    concept CEnumerable = std::ranges::range<TSelf>;
```

Sorry, I don't know the solution.
Do you have any ideas to solve it?

**Comment by FreePhoenix888** [comment]
@toge , Do you use C++20?

**Comment by FreePhoenix888** [comment]
@toge We have found out that apple-clang does not support ranges. We are going to drop that :)

**Pull request merged**